src/lib/viewers/ActionSummaryViewer.ts

```diff
@@ -10,6 +10,15 @@ abstract class AbstractActionSummaryViewer {
 		this.display = display;
 	}
 
+	public viewPreExecutionSummary(logFile: string): void {
```

Comment:
The `Display` object isn't directly accessible to the Action layer, so I opted to add the pre-execution output as another capability of the `ActionSummaryViewer` family instead of giving the Actions a new Dependency.